core/log-parser/log-readers/sftp.js

```diff
@@ -0,0 +1,39 @@
+import path from 'path';
+import SFTPTail from './sftptail.js';
+
+export default class TailLogReader {
+  constructor(queueLine, options = {}) {
+    for (const option of ['host', 'user', 'password', 'logDir'])
+      if (!(option in options)) throw new Error(`${option} must be specified.`);
+
+    this.reader = new SFTPTail({
+      host: options.host,
+      port: options.port || 22,
+      user: options.user,
+      password: options.password,
+      secure: options.secure || false,
+      timeout: options.timeout || 2000,
+      encoding: 'utf8',
+      verbose: options.verbose,
+
+      path: path.join(options.logDir, options.filename),
```

Comment:
It's weird but sometimes S/FTP module can't find the file due to reverse slashes `\`. It'd be the best to do a replace here.